editor/editor_settings.cpp

```diff
@@ -283,8 +283,9 @@ void EditorSettings::_load_defaults(Ref<ConfigFile> p_extra_config) {
 		hints["interface/editor/editor_language"] = PropertyInfo(Variant::STRING, "interface/editor/editor_language", PROPERTY_HINT_ENUM, lang_hint, PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_RESTART_IF_CHANGED);
 	}
 
-	_initial_set("interface/editor/hidpi_mode", 0);
-	hints["interface/editor/hidpi_mode"] = PropertyInfo(Variant::INT, "interface/editor/hidpi_mode", PROPERTY_HINT_ENUM, "Auto,VeryLoDPI,LoDPI,MidDPI,HiDPI", PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_RESTART_IF_CHANGED);
+	_initial_set("interface/editor/display_scale_auto", true);
+	_initial_set("interface/editor/display_scale", 1.0f);
+	hints["interface/editor/display_scale"] = PropertyInfo(Variant::REAL, "interface/editor/display_scale", PROPERTY_HINT_RANGE, "0.75,3,0.01", PROPERTY_USAGE_DEFAULT);
```

Comment:
We should keep `PROPERTY_USAGE_RESTART_IF_CHANGED` no? And likely add it for the new `display_scale_auto` parameter too.

Reply:
Changing the editor scale at runtime doesn't work; it has to be restarted, which means that `PROEPRTY_USAGE_RESTART_IF_CHANGED` has no practical effect here. (Despite its name, it does not actually prompt the user to restart the editor.)

Reply:
> (Despite its name, it does not actually prompt the user to restart the editor.)

What does it do then?

Reply:
I think all it does is restarting some logic in the editor, which is required to change things such as the editor font size. It's not a property that has "user-facing" consequences, but adding another property which would prompt the user to restart the editor could be useful.

Reply:
It indeed seems not to do anything. It was added in bccdc11dde but there's no code to use it.
It would still be better to keep it defined for the settings where its potential implementation would be useful, so that once we implement it (an issue should be opened to keep track of it), it will work.